docs/pages/reference/core/Button.mdx

```diff
@@ -191,26 +191,58 @@ button({
 The endpoint specified in `target` should return a JSON object with the following structure:
 
 ```ts
-type TransactionTargetResponse {
+type TransactionTargetResponse =
+  | TransactionTargetResponseSendTransaction
+  | TransactionTargetResponseSignTypedDataV4;
+
+type EthSendTransactionParams = {
+  /** JSON ABI. This must include the encoded function type and should include any potential error types. */
+  abi: JSON | Abi | [];
+  /** transaction to address */
+  to: Address;
+  /** value of ether to send with the transaction in wei */
+  value?: string;
+  /** optional transaction call data */
+  data?: Hex;
+};
+
+type TransactionTargetResponseSendTransaction = {
+  /** A [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md) chain ID to identify the tx network e.g. 'eip155:1' for Ethereum mainnet */
   chainId: string;
   method: "eth_sendTransaction";
-  attribution?: boolean;
+  /** Specific parameters for chainId and method */
   params: EthSendTransactionParams;
-}
-```
+  /** Return false to omit the [calldata attribution](https://www.notion.so/warpcast/Frame-Transactions-Public-9d9f9f4f527249519a41bd8d16165f73#c1c3182208ce4ae4a7ffa72129b9795a) suffix. If this value is undefined or true, clients will append the attribution suffix. */
+  attribution?: boolean;
+};
+
+type EthSignTypedDataV4Params = {
+  domain: {
+    chainId?: number;
+    name?: string;
+    salt?: Hex;
+    verifyingContract?: Address;
+    version?: string;
+  };
+  types: TypedData;
+  primaryType: string;
+  message: Record<string, unknown>;
+};
 
-```ts
-type EthSendTransactionParams {
-  abi: Abi | [];
-  to: Hex;
-  value?: string;
-  data?: Hex;
-}
+type TransactionTargetResponseSignTypedDataV4 = {
+  /** A [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md) chain ID to identify the tx network e.g. 'eip155:1' for Ethereum mainnet */
+  chainId: string;
+  method: "eth_signTypedData_v4";
+  /** Specific parameters for chainId and method */
+  params: EthSignTypedDataV4Params;
+  /** Return false to omit the [calldata attribution](https://www.notion.so/warpcast/Frame-Transactions-Public-9d9f9f4f527249519a41bd8d16165f73#c1c3182208ce4ae4a7ffa72129b9795a) suffix. If this value is undefined or true, clients will append the attribution suffix. */
+  attribution?: boolean;
```

Comment:
not applicable to signatures afaik - waiting for response to hear if it was a typo

Reply:
@michalkvasnicak attribution was removed from the spec